Detectors/TOF/base/src/Geo.cxx

```diff
@@ -19,3 +20,478 @@ constexpr Float_t Geo::HEIGHTS[NPLATES][NMAXNSTRIP];
 constexpr Float_t Geo::DISTANCES[NPLATES][NMAXNSTRIP];
 constexpr Bool_t Geo::FEAWITHMASKS[18];
 constexpr Float_t Geo::ROOF2PARAMETERS[3];
+
+void Geo::GetDetID(Float_t* pos, Int_t* det)
+{
+  //
+  // Returns Detector Indices (iSect,iPlate,iStrip,iPadX,iPadZ)
+  // space point coor (x,y,z) (cm)
+
+  det[0] = GetSector(pos);
+  det[1] = GetPlate(pos);
+  det[2] = GetStrip(pos);
+  det[3] = GetPadZ(pos);
+  det[4] = GetPadX(pos);
+}
+
+Int_t Geo::GetSector(const Float_t* pos)
+{
+  //
+  // Returns the Sector index
+  //
+
+  Int_t iSect = -1;
+
+  Float_t x = pos[0];
+  Float_t y = pos[1];
+  Float_t z = pos[2];
+
+  Float_t rho = TMath::Sqrt(x * x + y * y);
+
```

Comment:
You are comparing here rho (costs sqrt calculation) to two constants, would be more efficient to compare x*x+y*y to square of these constants (which are also constants)